Src/DependencyCollector/NetCore.Tests/DependencyTrackingTelemetryModuleTestNetCore.cs

```diff
@@ -0,0 +1,169 @@
+            }
+        }
+
+        /// <summary>
+        /// Tests that dependency is collected properly when there is no parent activity.
+        /// </summary>
+        [TestMethod]
+        [Timeout(5000)]
+        public async Task TestDependencyCollectionNoParentActivity()
+        {
+            ITelemetry sentTelemetry = null;
+
+            var channel = new StubTelemetryChannel
+            {
+                OnSend = telemetry =>
+                {
+                    // The correlation id lookup service also makes http call, just make sure we skip that
+                    DependencyTelemetry depTelemetry = telemetry as DependencyTelemetry;
+                    if (depTelemetry != null &&
+                        !depTelemetry.Data.StartsWith(FakeProfileApiEndpoint, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Assert.IsNull(sentTelemetry);
+                        sentTelemetry = telemetry;
+                    }
+                },
+                EndpointAddress = FakeProfileApiEndpoint
+            };
+
+            var config = new TelemetryConfiguration
+            {
+                InstrumentationKey = IKey,
+                TelemetryChannel = channel
+            };
+
+            config.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(config);
+
+                var url = new Uri("http://bing.com");
+
+                var request = new HttpRequestMessage(HttpMethod.Get, url);
+                await new HttpClient().SendAsync(request);
+
+                // on netcoreapp1.0 DiagnosticSource event is fired asycronously, let's wait for it 
+                Assert.IsTrue(SpinWait.SpinUntil(() => sentTelemetry != null, TimeSpan.FromSeconds(1)));
+
+                var item = (DependencyTelemetry)sentTelemetry;
+                Assert.AreEqual(url, item.Data);
+                Assert.AreEqual(url.Host, item.Target);
+                Assert.AreEqual("GET " + url.AbsolutePath, item.Name);
+                Assert.IsTrue(item.Duration > TimeSpan.FromMilliseconds(0), "Duration has to be positive");
+                Assert.AreEqual(RemoteDependencyConstants.HTTP, item.Type, "HttpAny has to be dependency kind as it includes http and azure calls");
+                Assert.IsTrue(
+                    DateTime.UtcNow.Subtract(item.Timestamp.UtcDateTime).TotalMilliseconds <
+                    TimeSpan.FromMinutes(1).TotalMilliseconds,
+                    "timestamp < now");
+                Assert.IsTrue(
+                    item.Timestamp.Subtract(DateTime.UtcNow).TotalMilliseconds >
+                    -TimeSpan.FromMinutes(1).TotalMilliseconds,
+                    "now - 1 min < timestamp");
+                Assert.AreEqual("200", item.ResultCode);
```

Comment:
I agree that we should use a local endpoint. 
Skipping status code validation is not an option since we also may get any kind of network issue.

Such tests become almost functional, however they are still quite small and run fast, so I would also prefer to keep them here, so it easier to detect issue.

There are 2 options
1. Do not fail if case of non 200 status OR any network issues (exception). Add small timeout, so test would not run for default 90 sec http timeout in case of error. Then functionality is still tested and rare issues do not make CI broken, however some implementation errors may be swallowed and since we are creating good pattern, it's not the best solution anyway.

2. Add local server with Anna or Nancy.
3. Change those tests to functional.

So, I would vote for 2nd option. What do you think?